test_conformance/relationals/test_comparisons_fp.h

```diff
@@ -0,0 +1,254 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef _TEST_COMPARISONS_FP_H
+#define _TEST_COMPARISONS_FP_H
+
+#include <vector>
+#include <map>
+#include <memory>
+#include <CL/cl_half.h>
+
+#include "testBase.h"
+
+#define HALF_NAN 0x7e00
+
+//--------------------------------------------------------------------------/
+
+template <typename T> using VerifyFunc = bool (*)(const T &, const T &);
+
+//--------------------------------------------------------------------------/
+
+using half = cl_half;
+
+//--------------------------------------------------------------------------
+
+struct RelTestBase
+{
+    RelTestBase(const ExplicitTypes &dt): dataType(dt) {}
+    ExplicitTypes dataType;
+};
+
+//--------------------------------------------------------------------------
+
+template <typename T> struct RelTestParams : public RelTestBase
+{
+    RelTestParams(const VerifyFunc<T> &vfn, const ExplicitTypes &dt,
+                  const T &nan_)
+        : RelTestBase(dt), verifyFn(vfn), nan(nan_)
+    {}
+
+    VerifyFunc<T> verifyFn;
+    T nan;
+};
+
+//--------------------------------------------------------------------------
+
+// Helper test fixture for constructing OpenCL objects used in testing
+// a variety of simple command-buffer enqueue scenarios.
+struct RelationalsFPTest
+{
+    RelationalsFPTest(cl_context context, cl_device_id device,
+                      cl_command_queue queue, const char *fn, const char *op);
+
+    virtual cl_int SetUp(int elements) = 0;
+
+    // Test body returning an OpenCL error code
+    virtual cl_int Run();
+
+    template <typename T>
+    void generate_equiv_test_data(T *, const unsigned int &, const bool &,
```

Comment:
```suggestion
    void generate_equiv_test_data(T *, unsigned int, bool,
```

Not sure why these are being passed as references?  Same for the next few methods.

Reply:
Done.